### PR TITLE
refactored PlayerRegistry and PlayerInfo

### PR DESCRIPTION
**Pull request by amazingalek**

### CODE REVIEW

File: QSB/PlayerRegistry.cs [hunk 1]
```diff
@@ -1,149 +1,46 @@
...
 using System.Linq;
-using UnityEngine;
+using QSB.TransformSync;
 
 namespace QSB
 {
     public static class PlayerRegistry
     {
-        public static List<PlayerInfo> PlayerList { get; } = new List<PlayerInfo>();
+        public static PlayerInfo LocalPlayer => GetPlayer(PlayerTransformSync.LocalInstance.netId.Value);
```
Comment (misternebula) [inline QSB/PlayerRegistry.cs:10]: does this run as soon as the script is loaded? because this will fail then

Comment (amazingalek) [reply QSB/PlayerRegistry.cs:10]: nope, when it's called

File: QSB/PlayerInfo.cs [hunk 2]
```diff
@@ -1,15 +1,54 @@
...
         public State State { get; set; }
+
+        public QSBTool Signalscope
+        {
+            get
+            {
+                var tools = Camera.GetComponentsInChildren<QSBTool>();
+                if (tools.Length == 0)
+                {
+                    DebugLog.ToConsole("Error - Zero items in QSBTool list while trying to get Signalscope", MessageType.Error);
```
Comment (amazingalek) [inline QSB/PlayerInfo.cs:27]: if the debug line can be removed this whole thing can be shortened to one line:
`public QSBTool Signalscope => Camera.GetComponentsInChildren<QSBTool>().FirstOrDefault(x => x.Type == ToolType.Signalscope);`

File: QSB/PlayerRegistry.cs [hunk 3]
```diff
@@ -1,149 +1,46 @@
...
-        public static Transform GetSector(uint id)
-        {
-            return GetPlayer(id).ReferenceSector;
-        }   
-
-        public static bool GetState(uint id, State state)
-        {
-            var states = GetPlayer(id).State;
-            return FlagsHelper.IsSet(states, state);
-        }
```
Comment (misternebula) [inline QSB/PlayerRegistry.cs:147]: Should this be moved into playerinfo?

Comment (amazingalek) [reply QSB/PlayerRegistry.cs:147]: i removed it as it wasn't used/needed

File: QSB/PlayerInfo.cs (hunk 2 above)
Comment (amazingalek) [reply QSB/PlayerInfo.cs]: i simplified it after testing it

File: QSB/State.cs [hunk 4]
```diff
@@ -8,7 +8,7 @@ public enum State
         Flashlight = 0,
         Suit = 1,
         ProbeLauncher = 2,
-        Signalscope = 4,
+        SignalScope = 4,
```
Comment (misternebula) [inline QSB/State.cs:11]: ?

**Pull request merged**